File: ulyp-ui/src/main/java/com/ulyp/ui/renderers/MultilinedText.java
package com.ulyp.ui.renderers;

import javafx.scene.text.Text;

public class MultilinedText extends Text {

    public MultilinedText(String value) {
        super(trimText(value));
    }

    static String trimText(String text) {
        if (text.length() < 100) {
            return text;
        }
        StringBuilder output = new StringBuilder(text.length() + 10);
        for (int i = 0; i < text.length(); i++) {
            if (i % 100 == 0) {
                output.append("\n");
            }
            output.append(text.charAt(i));
        }
        return output.toString();
    }
}

File: ulyp-ui/src/main/kotlin/com/ulyp/ui/renderers/MultilinedText.kt
package com.ulyp.ui.renderers

import javafx.scene.text.Text
import java.lang.StringBuilder

class MultilinedText(value: String) : Text(trimText(value)) {

    companion object {
        fun trimText(text: String): String {
            if (text.length < 100) {
                return text
            }
            val output = StringBuilder(text.length + 10)
            for (i in text.indices) {
                if (i % 100 == 0) {
                    output.append("\n")
                }
                output.append(text[i])
            }
            return output.toString()
        }
    }
}